ngmix/leastsqbound.py

```diff
@@ -6,12 +6,178 @@
 Placing a copy in this repo because that code is no longer maintained
 """
 
+import numpy as np
+from numpy import diag
+import logging
 import warnings
 
 from numpy import take, eye, triu, transpose, dot, finfo
 from numpy import empty_like, sqrt, cos, sin, arcsin, asarray
 from numpy import atleast_1d, shape, issubdtype, dtype, inexact
 from scipy.optimize import _minpack, leastsq
+from .util import print_pars
+from .flags import (
+    ZERO_DOF,
+    DIV_ZERO,
+    LM_SINGULAR_MATRIX,
+    LM_NEG_COV_EIG,
+    LM_NEG_COV_DIAG,
+    EIG_NOTFINITE,
+    LM_FUNC_NOTFINITE,
+)
+from .defaults import PDEF, CDEF
+
+LOGGER = logging.getLogger(__name__)
+
+
+def run_leastsq(func, guess, n_prior_pars, **keys):
+    """
+    run leastsq from scipy.optimize.  Deal with certain types of errors.  Allow
+    the user to input boundaries for parameters, dealt with in a robust way
+    using the leastsqbound code that wraps leastsq
+
+    Parameters
+    ----------
+    func:
+        the function to minimize
+    guess:
+        guess at pars
+    n_prior_pars:
+        number of slots in fdiff for priors
+    bounds : list, optional
+        ``(min, max)`` pairs for each element in ``x``, defining
+        the bounds on that parameter. Use None for one of ``min`` or
+        ``max`` when there is no bound in that direction.
+
+    some useful keywords
+    maxfev:
+        maximum number of function evaluations. e.g. 1000
+    epsfcn:
+        Step for jacobian estimation (derivatives). 1.0e-6
+    ftol:
+        Relative error desired in sum of squares, 1.0e06
+    xtol:
+        Relative error desired in solution. 1.0e-6
+    """
+
+    npars = guess.size
+    k_space = keys.pop("k_space", False)
+
+    res = {}
+    try:
+        lm_tup = leastsqbound(func, guess, full_output=1, **keys)
+
+        pars, pcov0, infodict, errmsg, ier = lm_tup
+
+        if ier == 0:
+            # wrong args, this is a bug
+            raise ValueError(errmsg)
+
+        flags = 0
+        if ier > 4:
+            flags = 2 ** (ier - 5)
```

Comment:
```suggestion
            flags |= 2 ** (ier - 5)
```

Reply:
in this case it doesn't matter but its good practice to always use `|=`